src/views/shared/amount-input/amount-input.view.jsx

```diff
@@ -92,12 +94,17 @@ function AmountInput(Component) {
      * @param {BigNumber} newAmount - New amount to be checked.
      */
     function checkAmountValidity(newAmount) {
-      const fee = getTransactionFee(transactionType, newAmount, account.token, l2Fee, gasPrice);
-      const newAmountWithFee = newAmount.add(fee);
+      const newFee = getTransactionFee(transactionType, newAmount, account.token, fee, gasPrice);
+      const newAmountWithFee = newAmount.add(newFee);
+      const isNewAmountMoreThanFunds = newAmountWithFee.gt(BigNumber.from(account.balance));
 
       setIsAmountNegative(newAmountWithFee.lte(BigNumber.from(0)));
```

Comment:
Yes I think that this check is old because is not posible to introduce negative values and it's better check newAmount. I kept this check because I don't know if it's posible to reach to that step with a negative value. Could do you think important refactor this part removing that logic or just change to newAmount?

Reply:
No, I would not remove this logic. Here we still have to check for negative values from the user. The regex is an extra security measure on the input, but we should not rely only on input validation.

All we need to do IMO is check the right value here, which is what the user inputs (`newAmount`) and not what the user inputs plus the fee (`newAmountWithFee`).

Checking if `newAmountWithFee` is negative is a bug.